production/src/synch/offering.rs

```diff
@@ -0,0 +1,107 @@
+use openapi::models;
+use openapi::apis::configuration::Configuration;
+use openapi::apis::class_schedules_api::v3_class_schedules_term_code_subject_catalog_number_get as get_class_schdule;
+use openapi::apis::class_schedules_api::V3ClassSchedulesTermCodeSubjectCatalogNumberGetError;
+use openapi::apis::Error;
+use super::sample_db_models;
+use tokio::task;
+use std::fs::{OpenOptions, File, create_dir_all};
+use std::path::Path;
+use std::io::{BufWriter, Write};
+use log::error;
+use stores::prelude::PgPool;
+
+pub async fn get_class_schedules_for_courses(
+    config: &Configuration,
+    term_code: &str,
+    courses: &Vec<sample_db_models::Course>
+) -> Vec<(i32, Vec<models::Class>)> {
+    
+    let mut schedule_data = Vec::<(i32, Vec<models::Class>)>::new();
+
+    // Make concurrent API calls for efficiency.
+    let task_results = courses.into_iter().map(|course| {
+        let config: Configuration = config.clone();
+        let term_code: String = term_code.to_string();
+        let id: i32 = course.id.clone();
+        let subject: String = course.subject_code.clone();
+        let catalog_number: String = course.catalog_number.clone();
+
+        task::spawn(async move {
+
+            let result: Result<
+                Vec<models::Class>, 
+                Error<V3ClassSchedulesTermCodeSubjectCatalogNumberGetError>
+            > = get_class_schdule(
+                &config, 
+                term_code.as_str(), 
+                subject.as_str(), 
+                catalog_number.as_str()).await;
+            (id, result)
+        })
+    });
+
+    for task in task_results {
+        let (id, result) = task.await.unwrap();
+        match result {
+            Ok(classes) => {
+                schedule_data.push((id, classes));
+            }
+            Err(_) => (),
+        }
+    }
+
+    return schedule_data;
+}
+
+pub fn map_course_ids_to_offering_query(course_ids: &Vec<i32>, term_name: String, file_path: &str) {
+    let v: Vec<&str> = term_name.split(' ').collect(); // Ex. term_name = "Spring 2023"
+    let term: String = String::from(v[0]);
+    let year: i32 = v[1].parse::<i32>().unwrap();
```

Comment:
nit: can either default to current year if empty or fail fast if didn't parse year correctly?